azure-pipelines.yml

```diff
@@ -0,0 +1,209 @@
+
+    - script: |
+        pushd ..
+        wget https://www.nxp.com/lgfiles/NMG/MAD/YOCTO/firmware-imx-7.9.bin
+        chmod +x firmware-imx-7.9.bin
+        yes | ./firmware-imx-7.9.bin
+        popd
+      displayName: Install NXP HDMI/DDR firmware
+
+    - script: |
+        pushd ../u-boot
+        export CROSS_COMPILE=$(Build.SourcesDirectory)/gcc-linaro-7.2.1-2017.11-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-
+        export ARCH=arm64
+        make imx8mq_evk_nt_defconfig
+        make
+        popd
+      displayName: Build U-Boot
+
+    - script: |
+        pushd ../imx-atf
+        export CROSS_COMPILE=$(Build.SourcesDirectory)/gcc-linaro-7.2.1-2017.11-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-
+        export ARCH=arm64
+        make PLAT=imx8mq SPD=opteed bl31
+        popd
+      displayName: Build ATF
+
+    - script: |
+        pushd ../optee_os
+        export -n CROSS_COMPILE
+        export -n ARCH
+        export CROSS_COMPILE64=$(Build.SourcesDirectory)/gcc-linaro-7.2.1-2017.11-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-
+        make PLATFORM=imx PLATFORM_FLAVOR=mx8mqevk CFG_TEE_CORE_DEBUG=n CFG_TEE_CORE_LOG_LEVEL=2  CFG_UART_BASE=0x30890000 CFG_RPMB_FS=y CFG_RPMB_TESTKEY=y CFG_RPMB_WRITE_KEY=y CFG_REE_FS=n CFG_IMXCRYPT=y CFG_CORE_HEAP_SIZE=98304
+        ${CROSS_COMPILE64}objcopy -O binary ./out/arm-plat-imx/core/tee.elf ./out/arm-plat-imx/tee.bin
+        popd
+      displayName: Build OP-TEE
+
+    - script: |
+        # make TAs!
+      displayName: Build TAs
+
+    - script: |
+        pushd ../imx-mkimage/iMX8M
+        export CROSS_COMPILE=$(Build.SourcesDirectory)/gcc-linaro-7.2.1-2017.11-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-
+        export ARCH=arm64
+        cp ../../firmware-imx-7.9/firmware/ddr/synopsys/lpddr4_pmu_train_*.bin .
+        cp ../../firmware-imx-7.9/firmware/hdmi/cadence/signed_hdmi_imx8m.bin .
+        cp ../../optee_os/out/arm-plat-imx/tee.bin .
+        cp ../../imx-atf/build/imx8mq/release/bl31.bin .
+        cp ../../u-boot/u-boot-nodtb.bin  .
+        cp ../../u-boot/spl/u-boot-spl.bin .
+        cp ../../u-boot/arch/arm/dts/fsl-imx8mq-evk.dtb .
+        cp ../../u-boot/tools/mkimage .
+        mv mkimage mkimage_uboot
+        cd ..
+        make SOC=iMX8M flash_hdmi_spl_uboot
+        popd
+      displayName: Build Imx-mkimage_uboot
+
+    - script: |
+        pushd ../mu_platform_nxp
+        export GCC5_AARCH64_PREFIX=$(Build.SourcesDirectory)/gcc-linaro-7.2.1-2017.11-x86_64_aarch64-linux-gnu/bin/aarch64-linux-gnu-
+        sudo python3.7 -m pip3 install -r requirements.txt --upgrade
+        python3.7 NXP/MCIMX8M_EVK_4GB/PlatformBuild.py --setup
+
+        cd MU_BASECORE
+        make -C BaseTools
+        cd ..
+
+        python3.7 NXP/MCIMX8M_EVK_4GB/PlatformBuild.py TOOL_CHAIN_TAG=GCC5 BUILDREPORTING=TRUE BUILDREPORT_TYPES="PCD" TARGET=RELEASE MAX_CONCURRENT_THREAD_NUMBER=20 BLD_*_CONFIG_NOT_SECURE_UEFI=1
+      displayName: Build UEFI
```

Comment:
Should there be a popd here?

Reply:
If anything the other pushd/popds are not needed. Each script task starts in the root directory (imx-iotcore checkout directory).